src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Asn1/RSAPrivateKeyAsn.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using System.Runtime.InteropServices;
+
+namespace System.Security.Cryptography.Asn1
+{
+    // https://tools.ietf.org/html/rfc3447#appendix-C
+    //
+    // RSAPrivateKey ::= SEQUENCE {
+    //   version           Version,
+    //   modulus           INTEGER,  -- n
+    //   publicExponent    INTEGER,  -- e
+    //   privateExponent   INTEGER,  -- d
+    //   prime1            INTEGER,  -- p
+    //   prime2            INTEGER,  -- q
+    //   exponent1         INTEGER,  -- d mod (p-1)
+    //   exponent2         INTEGER,  -- d mod (q-1)
+    //   coefficient       INTEGER,  -- (inverse of q) mod p
+    //   otherPrimeInfos   OtherPrimeInfos OPTIONAL
+    // }
+    //
+    // Version ::= INTEGER { two-prime(0), multi(1) }
+    //   (CONSTRAINED BY {
+    //     -- version must be multi if otherPrimeInfos present --
+    //   })
+    //
+    // Since we don't support otherPrimeInfos (Version=1) just don't map it in.
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct RSAPrivateKeyAsn
```

Comment:
ditto (unless the type name collides with something else then I'm ok with that)

Reply:
Yes, in general, the redundant Asn suffix is to avoid an existing conflict or a conceivable future conflict.